src/coreclr/scripts/superpmi_asmdiffs.py

```diff
@@ -0,0 +1,185 @@
+        print("Adding {} to PATH".format(jit_analyze_dir))
+        os.environ["PATH"] = jit_analyze_dir + os.pathsep + os.environ["PATH"]
+
+    # Find the portable `git` installation, and put `git.exe` on the PATH, for use by `jit-analyze`.
+    git_directory = os.path.join(script_dir, "git", "cmd")
+    git_exe_tool = os.path.join(git_directory, "git.exe")
+    if not os.path.isfile(git_exe_tool):
+        print("Error: `git` not found at {} (continuing)".format(git_exe_tool))
+    else:
+        # Put the git/cmd directory on the PATH so jit-analyze can find it.
+        print("Adding {} to PATH".format(git_directory))
+        os.environ["PATH"] = git_directory + os.pathsep + os.environ["PATH"]
+
+    # Figure out which JITs to use
+    os_name = "win" if platform_name.lower() == "windows" else "unix"
+    arch_name = coreclr_args.arch
+    host_arch_name = "x64" if arch_name.endswith("64") else "x86"
+    os_name = "universal" if arch_name.startswith("arm") else os_name
+    base_jit_path = os.path.join(coreclr_args.base_jit_directory, 'clrjit_{}_{}_{}.dll'.format(os_name, arch_name, host_arch_name))
+    diff_jit_path = os.path.join(coreclr_args.diff_jit_directory, 'clrjit_{}_{}_{}.dll'.format(os_name, arch_name, host_arch_name))
+
+    # Core_Root is where the superpmi tools (superpmi.exe, mcs.exe) are expected to be found.
+    # We pass the full path of the JITs to use as arguments.
+    core_root_dir = script_dir
+
+    print("Running superpmi.py download to get MCH files")
+
+    log_file = os.path.join(log_directory, "superpmi_download_{}_{}.log".format(platform_name, arch_name))
+    run_command([
+        python_path,
+        os.path.join(script_dir, "superpmi.py"),
+        "download",
+        "--no_progress",
+        "-core_root", core_root_dir,
+        "-target_os", platform_name,
+        "-target_arch", arch_name,
+        "-spmi_location", spmi_location,
+        "-log_level", "debug",
+        "-log_file", log_file
+        ], _exit_on_fail=True)
+
+    print("Running superpmi.py asmdiffs")
+    log_file = os.path.join(log_directory, "superpmi_{}_{}.log".format(platform_name, arch_name))
+
+    overall_md_summary_file = os.path.join(spmi_location, "diff_summary.md")
+    if os.path.isfile(overall_md_summary_file):
+        os.remove(overall_md_summary_file)
+
+    _, _, return_code = run_command([
+        python_path,
+        os.path.join(script_dir, "superpmi.py"),
+        "asmdiffs",
+        "--no_progress",
+        "-core_root", core_root_dir,
+        "-target_os", platform_name,
+        "-target_arch", arch_name,
+        "-arch", host_arch_name,
+        "-base_jit_path", base_jit_path,
+        "-diff_jit_path", diff_jit_path,
+        "-spmi_location", spmi_location,
+        "-error_limit", "100",
+        "-log_level", "debug",
+        "-log_file", log_file])
+
+    # If there are asm diffs, and jit-analyze ran, we'll get a diff_summary.md file in the spmi_location directory.
```

Comment:
Isn't that called `summary.md`?
https://github.com/dotnet/runtime/blob/dfb8c90b86ec67ffc1192a1561ce11e281ee9619/src/coreclr/scripts/superpmi.py#L2015-L2018

Reply:
Each MCH file creates a summary.md; superpmi.py appends these all together into a `diff_summary.md` file (so, one `diff_summary.md` file for each platform/arch run).